src/main/java/io/confluent/connect/jdbc/source/TimestampTableQuerier.java

```diff
@@ -132,7 +132,9 @@ public SourceRecord extractRecord() {
         || canCommitTimestamp(currentRecord.timestamp(), nextRecord.timestamp())) {
       latestCommittableTimestamp = currentRecord.timestamp();
     }
-    return currentRecord.record(latestCommittableTimestamp);
+    // set current in memory offset to the latestCommittableTimestamp
+    this.offset = new TimestampIncrementingOffset(latestCommittableTimestamp, null);
```

Comment:
Shouldn't this line be sitting under the if block above i.e. just below the line where we mutate latestCommittableTimestamp?

Reply:
>  Existing one would work too but seems we are creating the same object again and again unnecessarily when latestCommittableTimestamp does not change.

I agree, this is to keep things as they were already. Before this change as well we were updating offset each time - https://github.com/confluentinc/kafka-connect-jdbc/pull/1239/files#diff-304c376e74a018800a82cae4e42d9e58c60d6bb753c492ee3ed7e5297ba18910L151-L152 

Both are schematically equivalent from what I understand. I don't have a strong inclination toward either. I can move this to the if block above as well.

Reply:
@mukkachaitanya Grouping together related muations under one block helps understand the intent when someone is reading the code. However I don't know why previously the code was creating offset each time; just with a cursory peek at the code I din't find a reason for the same. But I have not checked if offset is thread-safe, is concurrent update fine etc. Do we have enough safety nests in terms of UTs/ITs/SystemTests for this connector? Is so moving the line inside the if and running the tests should be quicker; otherwise I am fine to leave it as it as because the suggestion is a minor one and doesn't weigh against regressions should safety nets are not robust for this Connector.

Reply:
Done!